### ISSUE TITLE
arrow-parens "as-needed" documentation is misleading

### ISSUE THREAD

**Issue by RobinDaugherty** [eslint/eslint#11202]
https://eslint.org/docs/rules/arrow-parens

Documentation states:

> "as-needed" allows omitting parens when there is only one argument.

However, it then gives an example of invalid syntax when using "as-needed":

> `a.then((foo) => {});`

**What did you expect to happen?**
Because the documentation says _allows_ omitting parens, it should not disallow parens around a single argument. So the following should both be valid:

```js
a.then((foo) => {});
a.then(foo => {});
```

While this is not valid:

```js
a.then(foo, bar => {});
```

**What actually happened? Please include the actual, raw output from ESLint.**

The rule _requires_ the parens to be omitted with a single argument.

```
Unexpected parentheses around single function argument (arrow-parens)
> 200 |     items.forEach((item) => {
```

**Are you willing to submit a pull request to fix this bug?**

Yes, subject to the project desires. My recommendation is to change the `as-needed` rule to _allow_ omitting parens, as described, and add another option (name to be determined) that required omitting the parens for single-argument functions.

**Comment by nzakas** [comment]
Thanks for the feedback. It seems like the problem here is with the documentation rather than with the code. The current behavior of `arrow-parens` matches what we do in other rules that use "as-needed" as an option, such as [`arrow-body-style`](https://eslint.org/docs/rules/arrow-body-style#as-needed).

I agree that the wording of the documentation is confusing and should be fixed.

**Comment by ljharb** [comment]
This has the accepted label, can it be reopened?

**Comment by lowderl** [comment]
Hi all,

I submitted the change for this. I just copied over the same language from the arrow-body-style docs as you mentioned @nzakas.
